src/main/kotlin/org/neo4j/graphql/handler/projection/ProjectionBase.kt

```diff
@@ -336,9 +348,27 @@ open class ProjectionBase {
         val relPattern = if (isRelFromType) "$childVariable:${relInfo.relType}" else ":${relInfo.relType}"
 
         val where = where(childVariable, fieldDefinition, nodeType, propertyArguments(field), field)
-        val fieldProjection = projectFields(childVariable, field, nodeType, env, variableSuffix)
 
-        val comprehension = "[($variable)$inArrow-[$relPattern]-$outArrow($endNodePattern)${where.query} | ${fieldProjection.query}]"
+        val orderBy = getOrderByArgs(field.arguments)
+        val sortByNeo4jTypeFields = orderBy
+            .filter { (property, _) -> nodeType.getFieldDefinition(property)?.isNeo4jType() == true }
+            .map { (property, _) -> property }
+            .toSet()
+
+        val fieldProjection = projectFields(childVariable, field, nodeType, env, variableSuffix, sortByNeo4jTypeFields)
+        var comprehension = "[($variable)$inArrow-[$relPattern]-$outArrow($endNodePattern)${where.query} | ${fieldProjection.query}]"
+        if (orderBy.isNotEmpty()) {
+            val sortArgs = orderBy.joinToString(", ", transform = { (property, direction) -> if (direction == Sort.ASC) "'^$property'" else "'$property'" })
+            comprehension = "apoc.coll.sortMulti($comprehension, [$sortArgs])"
+            if (sortByNeo4jTypeFields.isNotEmpty()) {
+                val neo4jFiledSelection = field.selectionSet.selections
+                    .filter { selection -> sortByNeo4jTypeFields.contains((selection as? Field)?.name) }
+                val deferredProjection = projectSelection("sortedElement", neo4jFiledSelection, nodeType, env, variableSuffix)
+                    .map { cypher -> cypher.query }
+                    .joinNonEmpty(", ")
+                comprehension = "[sortedElement IN $comprehension | sortedElement { .*, $deferredProjection }]"
```

Comment:
do we really want `.*` ? this can pull way too many properties that are not selected / asked for?
What do we need the deferredProjection for?

Reply:
Are those sort fields that are not in the projection?

Reply:
This logic I copied over form the js-library as well. The selection of the *. properties is done in the pattern comprehension so the `sortedElement` will already be _projected_